### DISCUSSION TITLE
quality-assurance-automation-ai

### DISCUSSION THREAD

**Discussion by giscus** [WorkSafeBC-Common-Engineering/Common.Engineering.Discussion#28] [Technical Blog]
# quality-assurance-automation-ai

https://wsbctechnicalblog.github.io/quality-assurance-automation-ai.html

**Comment by wpschaub** [comment]
Vikas, great post. 
Some community members are expressing the opinion that automation is not as valuable as it is made out to be, arguing that it is costly to upkeep and that a QA tester can conduct more comprehensive regression tests. Although I completely disagree with this view, I am interested in hearing your thoughts on the matter.

**Reply by vikasjawa** [reply]
It's understandable that some may question the value of automation in testing. While there is an initial investment required for setup, planning, and maintenance, the long-term benefits outweigh these costs. Automated testing enhances efficiency and ensures consistency.

I think automation is beneficial in regression testing, where hundreds of tests can be executed rapidly and consistently. This capability helps identify broken functionality early, potentially saving costs associated with bug fixes. 

Automation also empowers QA Engineers to concentrate their expertise on developing test strategies, plans, exploratory testing, and user experience testing. These tasks demand human judgment and domain knowledge, which are essential for thorough testing and quality assurance.

The key is to strike a balance between automated and manual testing based on project needs and resources.

**Reply by wpschaub** [reply]
Can you elaborate on what you believe is a healthy balance?

**Reply by vikasjawa** [reply]
I believe that achieving a balance should be driven by project-specific requirements and available resources. Few considerations that can help in striking the balance (or moving towards the achieving the balance):

- Determine suitable test cases for automated testing e.g., tests that require frequent execution such as regression tests following code changes.
- Determine suitable test cases for manual testing e.g., tests that require human judgment or exploration.
- If suitable then starting automation with high-impact tests or those prone to human error if executed manually. With the growing confidence we can expand the coverage gradually.

**Reply by AndreasNVI** [reply]
I wonder if you could leverage NLP to perform automated AI ad-hoc style testing? At least for the front-end systems...
